sdk/python/kfp/components/structures.py

```diff
@@ -74,60 +77,126 @@ class OutputSpec(base_model.BaseModel):
     description: Optional[str] = None
 
 
-class InputValuePlaceholder(base_model.BaseModel):
+class PlaceholderSerializationMixin(abc.ABC):
+    _FROM_PLACEHOLDER_REGEX: Union[str, type(NotImplemented)] = NotImplemented
+    _TO_PLACEHOLDER_TEMPLATE_STRING: Union[
+        str, type(NotImplemented)] = NotImplemented
+
+    @classmethod
+    def _is_input_placeholder(cls) -> bool:
+        field_names = {field.name for field in dataclasses.fields(cls)}
+        return "input_name" in field_names
+
+    @classmethod
+    def is_match(cls, placeholder: str) -> bool:
+        return re.match(cls._FROM_PLACEHOLDER_REGEX, placeholder) is not None
+
+    @classmethod
+    def from_placeholder(cls, placeholder: str) -> Any:
+        if cls._FROM_PLACEHOLDER_REGEX == NotImplemented:
+            raise NotImplementedError(
+                f'{cls.__name__} does not support placeholder parsing.')
+
+        matches = re.search(cls._FROM_PLACEHOLDER_REGEX, placeholder)
+        if matches is None:
+            raise ValueError(
+                f'Could not parse placeholder: {placeholder} into {cls.__name__}'
+            )
+        if cls._is_input_placeholder():
+            return cls(input_name=matches[1])
+        else:
+            return cls(output_name=matches[1])
+
+    def to_placeholder(self) -> str:
+        if self._TO_PLACEHOLDER_TEMPLATE_STRING == NotImplemented:
+            raise NotImplementedError(
+                f'{self.__class__.__name__} does not support creating placeholder strings.'
+            )
+        attr_name = 'input_name' if self._is_input_placeholder(
+        ) else 'output_name'
+        value = getattr(self, attr_name)
+        return self._TO_PLACEHOLDER_TEMPLATE_STRING.format(value)
+
+
+class InputValuePlaceholder(base_model.BaseModel,
+                            PlaceholderSerializationMixin):
     """Class that holds input value for conditional cases.
 
     Attributes:
         input_name: name of the input.
     """
     input_name: str
     _aliases = {'input_name': 'inputValue'}
+    _TO_PLACEHOLDER_TEMPLATE_STRING = "{{{{$.inputs.parameters['{}']}}}}"
+    _FROM_PLACEHOLDER_REGEX = r"\{\{\$\.inputs\.parameters\[(?:''|'|\")(.+?)(?:''|'|\")]\}\}"
 
 
-class InputPathPlaceholder(base_model.BaseModel):
+class InputPathPlaceholder(base_model.BaseModel, PlaceholderSerializationMixin):
     """Class that holds input path for conditional cases.
 
     Attributes:
         input_name: name of the input.
     """
     input_name: str
     _aliases = {'input_name': 'inputPath'}
```

Comment:
nit: I think we don't need this `_aliases` any more.

Reply:
Thanks for catching that, Chen. I agree we don't need this, but I think there is a tradeoff here. I'm curious what you think:

We need some way of calling (in this case, aliasing) the `input_name` attribute `inputPath` in order to support reading V1 YAML without any special handling. I think the best options for doing this are to either:
1) Keep the `_aliases` field (the approach above) or
2) Rename the `input_name` attribute to `inputPath`

I originally went with option 1 because calling the attribute `inputPath` is somewhat misleading, as it suggests the value of this attribute is the actual input path itself, rather than the *name* of the input path variable. I can see a case for either approach, however. WDYT?

Reply:
Definitely choose 1 over 2. Sorry again, I thought this was the aliase we used for Pydantic.